WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuElement.swift

```diff
@@ -15,30 +16,56 @@ struct HubMenuElement: View {
         Button {
             onTapGesture()
         } label: {
-            VStack {
-                ZStack {
-                    Color(UIColor(light: .listBackground,
-                                  dark: .secondaryButtonBackground))
-                    Image(uiImage: image)
-                        .renderingMode(.template)
-                        .resizable()
-                        .scaledToFit()
-                        .foregroundColor(Color(imageColor))
-                        .frame(width: iconSize, height: iconSize)
+            ZStack(alignment: .topTrailing) {
+
+                VStack {
+                    ZStack {
+                        Color(UIColor(light: .listBackground,
+                                      dark: .secondaryButtonBackground))
+                        Image(uiImage: image)
+                            .renderingMode(.template)
+                            .resizable()
+                            .scaledToFit()
+                            .foregroundColor(Color(imageColor))
+                            .frame(width: iconSize, height: iconSize)
+                    }
+                    .frame(width: imageSize, height: imageSize, alignment: .center)
+                    .cornerRadius(imageSize/2)
+                    .padding(.bottom, Constants.paddingBetweenElements)
+                    Text(text)
+                        .bodyStyle()
                 }
-                .frame(width: imageSize, height: imageSize, alignment: .center)
-                .cornerRadius(imageSize/2)
-                .padding(.bottom, Constants.paddingBetweenElements)
-                Text(text)
+                .frame(width: Constants.itemSize, height: Constants.itemSize)
+                HubMenuBadge(value: badge)
+                    .padding([.top, .trailing], 8)
+                    .renderedIf(badge > 0)
+            }
+        }
+    }
+
+    private struct HubMenuBadge: View {
+        let value: Int
+
+        var body: some View {
+            ZStack (alignment: .center) {
+                Rectangle()
+                    .fill(.purple)
+                    .cornerRadius(Constants.cornerRadius)
```

Comment:
nit: could we use `Circle` instead?

Reply:
I used Rectangle because in case the number of notifications has more than one digit, we need to expand the badge horizontally.